nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java

```diff
@@ -356,6 +356,162 @@ public void ternaryOperator() {
         .doTest();
   }
 
+  @Test
+  public void fieldAccessIndexArray() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  static @Nullable String [] fizz = {\"1\"};",
+            "  static final Integer i = 0;",
+            "  static void foo() {",
+            "  if (fizz[i]!=null) { ",
+            "   // field access indexes aren't handled",
+            "   // BUG: Diagnostic contains: dereferenced expression fizz[i] is @Nullable",
+            "   fizz[i].toString();",
```

Comment:
Hrm, I thought this case _would_ be handled?  What exact cases do we allow for index expressions again?

Reply:
Ah my bad, this should work fine for primitives. Updated this and method invocation case accordingly.